### DISCUSSION TITLE
Dark mode obscures checkbox checkmarks

### DISCUSSION THREAD

**Discussion by sebi-** [requarks/wiki#6150] [Error / Bug Report]
When using checklist - [ ] markdown and "appearance: dark" is enabled for the user, the checkmarks are (almost) not visible on macOS/Safari/Desktop. 

Steps to reproduce: 
1. Create a markdown page with this content:
- [x] item checked
- [ ] item not checked

2. Observe in the preview pane that the two items are almost indistinguishable (see screenshot attached).

<img width="183" alt="Screen Shot 2023-02-10 at 15 03 47" src="https://user-images.githubusercontent.com/1839356/218111156-90d202ca-0e00-4e57-8ca5-e0e1176d9392.png">

Possible fix (?): use a different color for dark mode on all
input type="checkbox" disabled="" checked="" class="task-list-item-checkbox"

**Comment by dr-waterstorm** [comment]
While it is not as bad on Linux - Chromium as it is in this screenshot, the colors should indeed be adjusted :+1: 

Suggesting this fix (We're currently using it with the CSS override):

```
.theme--dark .v-main .contents .task-list-item-checkbox:after {
  border-color: #e0e0e0;
  color: #e0e0e0
}
```

This sets the checkboxes to the same color as the text

**Comment by pallebone** [comment]
Hi there. 

I am also having this issue on mobile dark mode or light mode. The check in the checkbox is a very weak grey and hard to see. I tried adding this to my css override under ‘theme - code injection’

.theme--dark .v-main .contents .task-list-item-checkbox:after {
  border-color: #e0e0e0;
  color: #0f0
}

but it does nothing at all. 

Can someone help me to understand why adding this has no effect at all?

**Reply by dr-waterstorm** [reply]
Do you use the default theme?
What does the "Inspector" of the browser show? Does the CSS code show up?

I just tried your code snipped and it works perfectly fine in our setup:
![image](https://github.com/requarks/wiki/assets/11409264/6d299bad-ef07-484d-8b6d-d3ceb701174e)

Results in:
![image](https://github.com/requarks/wiki/assets/11409264/a3c1643b-8a6e-49e2-8d2f-b044baf332d0)

**Comment by pallebone** [comment]
Hello there @dr-waterstorm 

Thank you for your reply. My setup is very simple. I use https://www.pikapods.com/ which I believe you have a relationship with.

I have tested this in IOS (iphone) safari and also on the Edge browser on a windows machine (login with admin user).

I am not able to get this to work at this time :(

This is what my page looks like:
![image](https://github.com/requarks/wiki/assets/11403137/970b4d71-eefc-4dad-8e55-d820f3dcb07d)

The inspector has this:
![image](https://github.com/requarks/wiki/assets/11403137/e484f2be-072a-42f2-9c9e-660c89b47023)

One thing I am noticing is mine appear to be called "todo-list" while yours appear as "task-list"

How did you create your object so it was a task list? When I create the page in the editor the option along the top simply is this:
![image](https://github.com/requarks/wiki/assets/11403137/fe204e7d-9f54-40fa-a09c-d9fcb741f6db)

This could be the difference?

Kind regards
Peter

**Reply by dr-waterstorm** [reply]
Hi Peter,

yes, indeed that is the issue. We're using the markdown checkboxes. Those have a custom styling and can be modified quite easily.

You are using the visual editor. The checkboxes used there seem to be the "browser default" if I'm not mistaken. They look okay on my Computer. You can of course style them with CSS. 
However, if I understand correctly you basically need to replace the "default" checkboxes with custom ones in order to be able to change the color. Try searching for "CSS custom checkbox color". See for example the [W3School](https://www.w3schools.com/howto/howto_css_custom_checkbox.asp)

You can target the "To-do List" checkboxes with:

```
.theme--dark .v-main .contents .todo-list .todo-list__label input {

}
```

**Comment by pallebone** [comment]
Im sorry I dont know how to do what you are saying, which is why I use the simple editor. I tried following the link but could not complete what you were asking me to do. Is there no way to make the normal editor use the different checkboxes that you use?

**Comment by pallebone** [comment]
Sadly only way I could do anything readable was add in a symbol on top of the checkbox such as:

.theme--dark .v-main .contents .todo-list .todo-list__label input:checked:after {
  content: '＇';
  color: #00f;
  font-weight: 900
}

Makes it readable at least. Strange that dokuwiki uses a different kind of checkbox depending on the editor, causing this problem.
Giving up and leaving it as this.

**Comment by dr-waterstorm** [comment]
Yes, that is sadly more complicated.
I think you need to completely style the boxes yourself to get the effect you want, because by default, the browser generates the boxes and therefore the style depends on your browser.

We use markdown to create pages. This can be selected when creating  a new page. This is a different way of creating pages and does not give you a WYSIWYG editor. Just a more basic editor with some help. It seems that markdown checkboxes have their own styling and there do not use the browser default.

You can achieve something like that with the following code. NOTE: This is only for dark mode. In light mode the checkboxes still use the browser default and therefore will look different.

```
.theme--dark .v-main .contents .todo-list .todo-list__label input[disabled] {
    width: 1.1rem;
    height: 1.1rem;
    top: 2px;
    position: relative;
    margin-right: 4px;
}

.theme--dark .v-main .contents .todo-list .todo-list__label input[disabled][checked]:after {
    content: "✓";
}

.theme--dark .v-main .contents .todo-list .todo-list__label input[disabled]:after {
    position: absolute;
    left: 0;
    top: 0;
    content: " ";
    display: block;
    width: 1.1rem;
    height: 1.1rem;
    background-color: #121212;
    border-color: #e0e0e0;
    color: #0f0;
    border: 1px solid #bdbdbd;
    border-radius: 2px;
    font-weight: 700;
    font-size: .8rem;
    line-height: 1rem;
    text-align: center;
}
```

That will give you something like this:
![image](https://github.com/requarks/wiki/assets/11409264/a2eb1696-6844-4876-bb2b-a7ee99f6c960)

**Comment by pallebone** [comment]
@dr-waterstorm Thank you for this. I just tested it and it works perfectly.
I still think its odd that the 2 different editors create different kinds of checkboxes instead of there being uniformity across the platform but this workaround seems totally acceptable and the users of the system will be able to use it in this way.

Thanks for the help :)

P
